### PR TITLE
Remove WebGPU support

### LINKED ISSUE #4101: Remove WebGPU env support from SPIRV-Tools
Remove all of the functional code related to this execution environment.

### PR DESCRIPTION
**Pull request by zoddicus**
Leaves SPV_ENV_WEBGPU_0 enum in place, but marked deprecated, so users
of the library are not broken by an API enum being removed.

Fixes #4101

### CODE REVIEW

**Comment by zoddicus** [comment]
PTAL, the shaderc-smoketest failure appears to be some gtest weirdness unrelated to my patch.

File: source/val/validate_builtins.cpp [hunk 1]
```diff
@@ -3768,26 +3667,17 @@ spv_result_t BuiltInsValidator::ValidateSingleBuiltInAtDefinition(
...
-    // WebGPU spec.
+  if (!spvIsVulkanEnv(_.context()->target_env)) {
+    // Early return. All currently implemented rules are based on Vulkan spec.
     //
     // TODO: If you are adding validation rules for environments other than
-    // Vulkan or WebGPU (or general rules which are not environment
-    // independent), then you need to modify or remove this condition. Consider
-    // also adding early returns into BuiltIn-specific rules, so that the system
-    // doesn't spawn new rules which don't do anything.
+    // Vulkan (or general rules which are not environment  independent), then
```
Comment (alan-baker) [inline source/val/validate_builtins.cpp:3674]: Nit: appears to be two spaces after environment.

**Review by alan-baker — CHANGES_REQUESTED** [review]
For the command line tools at least it might make sense to issue an error if `--target-env=webgpu` is used. It seems to assert in debug, but a clear error would be more helpful.

File: source/val/validate_builtins.cpp (hunk 1 above)
Comment (zoddicus) [reply source/val/validate_builtins.cpp:3674]: Done

**Comment by zoddicus** [comment]
> For the command line tools at least it might make sense to issue an error if `--target-env=webgpu` is used. It seems to assert in debug, but a clear error would be more helpful.

webgpu0 was the old string and it using it for '--target-env=' give the following error:
error: Invalid value passed to --target-env

Are you wanting a more specific error about WebGPU no longer being implemented or is that sufficient?

**Comment by alan-baker** [comment]
> > For the command line tools at least it might make sense to issue an error if `--target-env=webgpu` is used. It seems to assert in debug, but a clear error would be more helpful.
> 
> webgpu0 was the old string and it using it for '--target-env=' give the following error:
> error: Invalid value passed to --target-env
> 
> Are you wanting a more specific error about WebGPU no longer being implemented or is that sufficient?

No, that's good. I guess I missed that.

**Review by alan-baker — APPROVED** [review]

**Pull request merged**